File: eAutoShop/eAutoShop.UI/eautoshop_desktop/lib/models/employee_task/employee_task_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'employee_task_search_object.g.dart';

@JsonSerializable(includeIfNull: false)
class EmployeeTaskSearchObject {
  final int? page;
  final int? pageSize;
  final String? titleFTS;
  final int? employeeId;
  final int? createdById;
  final String? state;
  final DateTime? minDueDate;
  final DateTime? maxDueDate;

  const EmployeeTaskSearchObject({
    this.page,
    this.pageSize,
    this.titleFTS,
    this.employeeId,
    this.createdById,
    this.state,
    this.minDueDate,
    this.maxDueDate,
  });

  factory EmployeeTaskSearchObject.fromJson(Map<String, dynamic> json) =>
      _$EmployeeTaskSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeTaskSearchObjectToJson(this);
}

File: eAutoShop/eAutoShop.Model/SearchObjects/EmployeeTaskSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eAutoShop.Model.SearchObjects
{
    public class EmployeeTaskSearchObject : BaseSearchObject
    {
        public string? TitleFTS { get; set; }

        public int? EmployeeId { get; set; }

        public int? CreatedById { get; set; }

        public string? State { get; set; }

        public DateTime? MinDueDate { get; set; }

        public DateTime? MaxDueDate { get; set; }
    }
}
